### PR TITLE
feat: maybe and either

### PR DESCRIPTION
**Pull request by adrizein**
Hi, thank you for maintaining this package !
This is an attempt to implement #4 : Maybe and Either

### CODE REVIEW

**Comment by galkowskit** [comment]
Thanks! It's a library that I use personally for playing around with Deno. :P
I'll take a look at the PR after work but from what I have seen it looks pretty nice.

CI matrix tests for Deno 0.32.0 and 0.31.0 fail because `Deno.test` was introduced in `0.33.0`. I have to fix that. 🤔

**Comment by galkowskit** [comment]
Hey! I had some time to play around with those. I think these are great and I think they'll be a great fit. There are a couple of small things that I was thinking about adding (now or later as separate issues).

One is the usage of `[Deno.symbols.customInspect]` for easier debugging, especially for people less accustomed to functional programming and who want to just `console.log` stuff and see what it is.

For example in `Maybe`:
```typescript
// ...
mapMaybe<R>(f: (t: T) => Maybe<R>): Maybe<R>;

    toJSON(): T | undefined;
    toString(): string;
    [Deno.symbols.customInspect]: () => string;
}
```

And then in `just`:
```typescript
// ...
*[Symbol.iterator]() {
    yield value;
},
[Deno.symbols.customInspect]() {
    return `just{${value}}`
}
```

Another thing that I noticed (but that's something that we probably won't be able to overcome in current TypeScript) is that in some cases `.map` on Maybe will lose typings and default to `unknown` so `<number>` cast (for example) is required.

```typescript
const double = (x: number) => x * 2;

const input: any = [5, 2, 1, 5, 7];

const res = maybe(input)
    .map(reverse)
    .map<number>(head)
    .map(double)
    .default(0);
```

Lastly, there are some minor README duplications that I will mark in the code. :)

File: README.md [hunk 1]
```diff
@@ -91,6 +91,140 @@ const filterAlfas = filterCarsByMake("Alfa Romeo");
...
+```
+
+### either.error
+**either.error** wraps a value in the Either interface as a right value.
+`JSON.stringify()` will throw the value (ideally an Error)
+```typescript
+import either from "https://deno.land/x/denofun/lib/either.ts";
+// import error from "https://deno.land/x/denofun/lib/either/error.ts"; // is also available
+
+const numberOrError = either.jsonError<number, Error>(new Error("message"));
```
Comment (galkowskit) [inline README.md]: Both examples for `error` and `jsonError` use `jsonError` function.

Comment (adrizein) [reply README.md]: fixed :)

**Comment by adrizein** [comment]
@galkowskit is this ok for you ?

**Comment by galkowskit** [comment]
Yup, great job! Looks good to me, I'll squash and merge it. And I'll fix the CI this evening so it doesn't fail on 0.32.0. :)

**Comment by adrizein** [comment]
🎉

**Pull request merged**